src/isar/services/service_connections/echo/echo_service.py

```diff
@@ -180,23 +173,21 @@ def get_mission(self, mission_id: int) -> Optional[Mission]:
         """
         mission_plan: Optional[dict] = self.mission_plan(mission_id)
 
-        if mission_plan is None:
-            return None
         mission_tags: List[dict] = mission_plan["planItems"]
         mission: Mission = Mission(mission_steps=[])
 
         for tag in mission_tags:
             tag_name: str = tag["tag"]
 
-            drive_step: Optional[DriveToPose] = self.create_drive_step(
-                tag_name=tag_name
-            )
-
-            measurement_steps: Optional[
-                List[Union[TakeImage, TakeThermalImage]]
-            ] = self.create_measurement_steps(tag_name=tag_name)
+            try:
+                drive_step: Optional[DriveToPose] = self.create_drive_step(
+                    tag_name=tag_name
+                )
```

Comment:
The `create_drive_step` can return None but only because it uses the old return None functionality rather than raise an exception. See line 103 in this file.

This function should also cast an exception rather than returning None.